android/src/test/java/com/rudderstack/android/internal/infrastructure/LifecycleObserverPluginTest.kt

```diff
@@ -16,5 +91,117 @@ class LifecycleObserverPluginTest {
         lifecycleObserverPlugin.onAppBackgrounded()
 
         verify { analytics.flush() }
+        lifecycleObserverPlugin.shutdown()
```

Comment:
Is it necessary to call this shutdown()? 
Because the `lifecycleObserverPlugin` object is a local variable, I think it would not make any difference to other tests if we don't call this shutdown(). Also after calling shutdown, we are not verifying any action. So please consider removing it.

If you think, it's mandatory then let's move this shutdown() call before `verify`.

Reply:
If you have some other thoughts about keeping this, please let me know.

Reply:
Shutdown should be called on all plugins that call setup. Otherwise it might leak. May not be with this implementation, but in future changes